platform/src/components/cloudflare/binding.ts

```diff
@@ -78,3 +90,38 @@ export function binding<T extends Binding["type"]>(input: Binding & {}) {
     properties: input.properties as Extract<Binding, { type: T }>["properties"],
   };
 }
+
+/**
+ * @param bindingName - The name of the global variable for the binding in your Worker code.
+ * @param cfBinding
+ */
+export function toWorkerScriptBinding(bindingName: Input<string>, cfBinding: ReturnType<typeof binding>): WorkerScriptKvNamespaceBinding | WorkerScriptServiceBinding | WorkerScriptQueueBinding | WorkerScriptR2BucketBinding | WorkerScriptD1DatabaseBinding {
```

Comment:
@fwang, I'm not sure what convention you have within the team. This could either be solved more elegantly (without type casting) by playing around with generics and relying more on types, or loosen the types a bit and add regression tests. I could try to solve it in both ways, but I would like some validation from your side

Reply:
I see that @thdxr was assigned, then I would like to forward the question to you.